tox-offline-draft.md

```diff
@@ -5,77 +5,154 @@
 This proposal aims to provide a general and technical walkthrough of an
 offline messaging protocol for use with the Tox protocol using its existing
 Distributed Hash Table (DHT). Messages are only stored in the DHT when the
-message receiver is currently offline. The sender does not have to be
-online for the receiver to retrieve messages.
+message receiver is currently offline. The sender does not have to be online
+for the receiver to retrieve messages.
 
 As this method uses the DHT, it is currently UDP only and will not work
 unless UDP is enabled. This restriction will hopefully be lifted with
 further work and development of the Tox protocol.
 
+## Terminology
+PoW:
+  Proof-of-Work -- Determines CPU work necessary in order to store an offline message on the DHT.
+
+SENDER:
+  The friend of RECEIVER sending an offline message to her when she is offline.
+
+RECEIVER:
+  The friend of SENDER receiving an offline message.
+
+STORAGE_NODE:
+  One of potentially many nodes storing offline messages for other users.
+
+OM Keys:
+  Offline Messaging public / private keypair.
+
 ## DHT Extension
 
-The DHT must be extended in order to be able to store arbitrary data. Tox
-Clients should have a reasonable default for DHT storage size set and
+The DHT must be extended in order to be able to store offline messages. Tox
+Clients should have a reasonable default set for DHT storage size and
 optionally allow users to change it. This space will be used for storing
 messages of other users.
 
-## Inserting Messages 
+## Storage Space Allocation
 
-Analagous to dpush [0], a user creates a new public and private key pair
-(Current DHT keys are ephemeral) to sign their inbox location. The inbox is
-stored in the DHT on computers who are in charge of that key range. (TODO:
-Should the public key be hashed?) Users then store messages they want to
-send to a store in that inbox's location. 
+The RECEIVER must request a certain amount of storage space for incoming
+offline messages on a number of STORAGE_NODEs. A PoW job must be completed
+by the RECEIVER to do this. The more space required per node, the higher
+the PoW difficulty.
```

Comment:
How complex PoW do you want to calculate for reasonable amount of storage?

I was thinking to have ~1 second PoW per average message on average device. Do you think it would be too expensive to generate it per message?

If you would like to pre-generate storage for 50 average messages it would take you ~50 seconds (in case we want to use PoW with same complexity in total). In case you have 10 friends online, you have to spent ~500 seconds to pre-generate all required storage space. Spreading calculations and calculating PoW per message looks better to me.